[17.0][MIG] account_banking_sepa_direct_debit

JasminSForgeFlow opened the pull request:
Standard Migration

@FrogeFlow

Depends on:
- https://github.com/OCA/bank-payment/pull/1231

Commit:
Add module account_banking_sepa_direct_debit that implements pain.008.001.02, pain.008.001.03 and pain.008.001.04. This module is not ready yet : the management of mandates is still missing. I am currently trying to get more information about these mandates to decide what is the best implemtation of the data model of the mandates (O2M on res.partner ? O2M os res.partner.bank ?).

Commit:
Launchpad automatic translations update.

pedrobaeza committed:
[MIG] account_banking_sepa_direct_debit: Migration to 8.0

Commit:
replace old sdd.mandate referencess

pedrobaeza committed:
[FIX+IMP] account_banking_sepa_direct_debit: Several things:

* Restore scheme field in mandate view
* Uncaught flake8 errors
* Set args correctly for cron job
* Split for removing stock dependency on sale
* Remove old translations templates
* More reorder on folder structure
* Translation template files
* Translations to spanish

Commit:
[IMP] account_banking_sepa_direct_debit: 5 things:

* Transfer move: one transfer move for each payment.order with only 1 line in the transfer account for the total of the account move. Move the inherit of the 'Invoice' button of payment.order from account_banking_payment_transfer to account_banking_payment_export Demo data: Add a bank account + mandate on Agrolait, to be able to easily test multi-partner SEPA DD
* mandates: search by reference, add group_by, add seq type in tree view
* Add scheme in mandate tree+search view Replace tabs by spaces in mandate views account_banking_mandate/views/account_banking_mandate_view.xml: convert from dos to unix format
* Add sale_ok and purchase_ok filters in partner/sale/purchase form views Use widget=selection for payment_mode_id fields Update demo data
* Limit lenght of mandate reference, according to ISO 20022

sbidoul committed:
sepa direct debit: get BIC from bank account if not available on bank

Commit:
Use attachments instead of creating a specific object for each kind of payment order

phschmidt committed:
Add requested descriptions in an OCA README.rst file

alexis-via committed:
Port SEPA modules to new API

Fix an important regression in account_banking_sepa_direct_debit: "Date of Last Debit" was not set any more
Proper write of date_done with account_banking_payment_export is installed without account_banking_payment_transfer
Add post-install script for date_sent on payment.order

sergio-teruel committed:
account_banking_sepa_direct_debit: Fix error in payment order validation when there is more than one line with the same mandate.

alexis-via committed:
[IMP] account_banking_sepa_direct_debit: 4 things

* Add bank.payment.lines object to allow grouping in the payments
* Add a hook to inherit grouping of the transfer account move line

  Use that new hook in SEPA direct debits
  Better variable names
* FIX Reading wrong field for sequence type of SEPA DD
  Loop on bank payment lines instead of payment lines
* Update automated tests and demo data
* Add on_change on field 'type' of payment.mode for easier configuration

pedrobaeza committed:
[IMP] account_banking_sepa_direct_debit: Short headers + Bump version numbers + es translations

antespi committed:
Define SEPA identifiers per payment mode

pedrobaeza committed:
account_banking_sepa_direct_debit: Fixes #257

sergio-teruel committed:
sepa_direct_debit_mandate: Split basic and sepa mandate

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
[MIG] account_banking_sepa_direct_debit: Migration to 9.0

pedrobaeza committed:
[FIX+IMP] account_banking_sepa_direct_debit: 3 things:

* Remove unused POT files
* Spanish translation
* Fix mandate report layout

oca-transbot committed:
OCA Transbot updated translations from Transifex

antespi committed:
[IMP] account_banking_sepa_direct_debit: Add english lang to mandate

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[MIG] account_banking_sepa_direct_debit: Migration to 10.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
account_banking*: Show identifiers at payment mode level

Now the visibility is controlled through a security group.

etobella committed:
[MIG] account_banking_sepa_direct_debit

oca-transbot committed:
OCA Transbot updated translations from Transifex

astirpe committed:
[11.0][FIX] account_banking_sepa_direct_debit: printing report

pedrobaeza committed:
[FIX] account_banking_sepa_direct_debit: Fix tests due to upstream change

Odoo has added a constraint for avoiding a company currency change if there
are move lines, making these tests to fail, as the currency is changed to EUR.

With this commit, we create a new company with EUR currency for avoiding the
problem.

This commit also changes account_banking_mandate for not duplicating mandate
number, as it was detected during the test creation.

Similar to 1f8e345469695d1fb1a2ba2109ddea3adbdf1f78

Commit:
[10.0][FIX] Split account move generated for SEPA direct debits per sequence_type (#464)

The sequence_type of mandates switches from First to Recurring:
- BEFORE: when state switched from 'open' to 'generated'
- NOW: when state switches from 'generated' to 'uploaded'

sbidoul committed:
[ADD] sepa: support for Purpose Code

etobella committed:
Translated using Weblate (Español (España))

Currently translated at 100,0% (60 of 60 strings)

Translation: bank-payment-11.0/bank-payment-11.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-11-0/bank-payment-11-0-account_banking_sepa_direct_debit/es/

andreparames committed:
[ADD] sepa: add Instruction Identification field

InstrId is optional, so there should be no harm in
adding it always, with the same value as EndToEndId.

sergio-teruel committed:
[MIG] account_banking_sepa_direct_debit: Migration to 12.0

pedrobaeza committed:
[MIG] account_banking_sepa_direct_debit: Finish migration to 12.0

* Fixed tests
* Adjust code for Python dates handling
* Update copyrights

ThomasBinsfeld committed:
[FIX] Account Banking SEPA Direct Debit: creditor identifier in config
The sepa_creditor_identifier in res.config.settings is readonly by default (related) but should not.

ThomasBinsfeld committed:
[REF] Account Banking SEPA Direct Debit: check SEPA ready

sergio-teruel committed:
[12.0][FIX] account_banking_sepa_direct_debit: Fix positional arguments in message_post

Commit:
Added translation using Weblate (Italian)

Commit:
Translated using Weblate (Italian)

Currently translated at 22.1% (17 of 77 strings)

Translation: bank-payment-12.0/bank-payment-12.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-12-0/bank-payment-12-0-account_banking_sepa_direct_debit/it/

pedrobaeza committed:
[IMP] account_banking_sepa_direct_debit: Rise exception if the mandate type is not correct

Closes #600

sergio-teruel committed:
[FIX] account_banking_sepa_direct_debit: Fix report mandate. Adapt to bootstrap 4

oihane committed:
Translated using Weblate (Spanish)

Currently translated at 75.6% (59 of 78 strings)

Translation: bank-payment-12.0/bank-payment-12.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-12-0/bank-payment-12-0-account_banking_sepa_direct_debit/es/

misern2 committed:
[IMP] account_banking_sepa_direct_debit: black, isort

misern2 committed:
[MIG] account_banking_sepa_direct_debit: Migration to 13.0

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (71 of 71 strings)

Translation: bank-payment-13.0/bank-payment-13.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-13-0/bank-payment-13-0-account_banking_sepa_direct_debit/es_AR/

pedrobaeza committed:
[IMP] account_banking_sepa_direct_debit: Ease inheritance in tests

- Unfold SDD test in 2 classes: the base with initialization + the other with tests
- Internal variable for parametrizing the chart template to load

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (71 of 71 strings)

Translation: bank-payment-13.0/bank-payment-13.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-13-0/bank-payment-13-0-account_banking_sepa_direct_debit/es_AR/

pedrobaeza committed:
[IMP] account_banking_sepa_direct_debit: Remove l10n_generic_coa hidden dependency in tests

As we were setting a new company with its CoA, we need to load in it one, but using
l10n_generic_coa doesn't serve, as we don't depend on that module (and we shouldn't),
but we are invoking its XML-ID for loading the CoA, trusting in the good faith that
the module is available and installed, which is not the case in some cases.

Luckily, Odoo provides already an inheritable test for having a minimal CoA that
doesn't require to load any chart. This PR uses that test and performs the minimal
changes to adapt the module tests to it.

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (71 of 71 strings)

Translation: bank-payment-13.0/bank-payment-13.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-13-0/bank-payment-13-0-account_banking_sepa_direct_debit/es_AR/

CarlosRoca13 committed:
[IMP] accont_banking_sepa_direct_debit: black, isort, prettier

Commit:
[MIG] account_banking_sepa_direct_debit to v14.0

CarlosRoca13 committed:
[FIX] account_banking_sepa_direct_debit: Fixed settings view and test_sdd implementation

[UPD] Update account_banking_sepa_direct_debit.pot

[UPD] README.rst

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 100.0% (71 of 71 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_banking_sepa_direct_debit/fr/

alexis-via committed:
account_banking_sepa_direct_debit: fix print_report_name

alexis-via committed:
Add print button on mandate

[UPD] Update account_banking_sepa_direct_debit.pot

account_banking_sepa_direct_debit 14.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-14.0/bank-payment-14.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_banking_sepa_direct_debit/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (77 of 77 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_banking_sepa_direct_debit/es_AR/

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 89.6% (69 of 77 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_banking_sepa_direct_debit/nl/

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 92.2% (71 of 77 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_banking_sepa_direct_debit/nl/

alexis-via committed:
[FIX] account_banking_sepa_direct_debit: fix crash when printing a mandate with empty partner

alexis-via committed:
Update test suite

Fix bug in account_banking_mandate when a res.partner.bank is not attached to a particular company (company_id is NOT required on res.partner.bank)

oca-travis committed:
[UPD] Update account_banking_sepa_direct_debit.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-14.0/bank-payment-14.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_banking_sepa_direct_debit/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (77 of 77 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_banking_sepa_direct_debit/es_AR/

etobella committed:
[FIX] account_banking_sepa_direct_debit: Bank Account and Mandate should not be required

The bank account is taken from the mandate. Mandate is autopopulated if not found.

alexis-via committed:
sepa_direct_debit: mandate_required back to True

sepa_credit_transfer: bank_account_required back to True
Partial revert of the breakage of PR #875

account_banking_sepa_direct_debit 14.0.1.3.0

Shide committed:
[FIX] [14.0] Removed external dependencies already defined in Odoo base

account_banking_sepa_direct_debit 14.0.1.3.1

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

manuelregidor committed:
[MIG]account_banking_sepa_direct_debit: Migration to 15.0

[UPD] Update account_banking_sepa_direct_debit.pot

[UPD] README.rst

ramiadavid committed:
[FIX] account_banking_pain_base, account_banking_sepa_credit_transfer, account_banking_direct_debit: Remove Environment.manage()

Environment.manage() is deprecated in v 15.0

OCA-git-bot committed:
account_banking_sepa_direct_debit 15.0.1.0.1

pedrobaeza committed:
[REF] account_banking_sepa_direct_debit: Adapt module to native payment refactoring

Notable changes:

- InstrId and EndToEndId tags are now filled with the journal entry ID,
  as there's no bank payment line identifier.

gfcapalbo committed:
[IMP] account_banking_sepa_direct_debit: pre-commit stuff

gfcapalbo committed:
[IMP] redo readme 16.0

Reyes4711-S73 committed:
[16.0][MIG] account_banking_sepa_direct_debit: Migration to 16.0

pedrobaeza committed:
[FIX] account_banking_sepa_*: Remove post-installs

From this version, Odoo already includes code on the
account.payment.method creation for adding such method to each journal:

https://github.com/odoo/odoo/blob/a93ed75cb80f16f/addons/account/models/account_payment_method.py#L24-L38

so we don't need to explicitly create them for the new payment methods
we are adding in these modules.

TT41146

stefan-tecnativa committed:
[FIX] account_banking_sepa_direct_debit: CSS styling was not being applied in reports

account_banking_sepa_direct_debit 15.0.2.1.0

Commit:
[UPD] Update account_banking_sepa_direct_debit.pot

OCA-git-bot committed:
[UPD] README.rst

Gelo-fl committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (77 of 77 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_sepa_direct_debit/es/

remi-filament committed:
Translated using Weblate (French)

Currently translated at 100.0% (77 of 77 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_sepa_direct_debit/fr/

lmignon committed:
[FIX] account_banking_sepa_direct_debit: Avoid singleton error

OCA-git-bot committed:
account_banking_sepa_direct_debit 16.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

alexis-via committed:
Add warning when payment order is not SEPA

The field 'sepa' on account.payment.order is only display for SEPA
payment methods.
If the option "show warning if not SEPA" is enabled on the payment
method, a warning banner is now displayed on payment orders with a SEPA
payment method which are not SEPA.

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_sepa_direct_debit/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (74 of 74 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_sepa_direct_debit/es/

stefan-tecnativa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (74 of 74 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_sepa_direct_debit/es/

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 22.9% (17 of 74 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_sepa_direct_debit/it/

MiquelRForgeFlow commented on account_banking_sepa_direct_debit/data/mandate_expire_cron.xml:
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8" ?>
+<?xml version="1.0" ?>
```
Please @JasminSForgeFlow revert this kind of change in the .xml files. The warning is for .html files.

JasminSForgeFlow replied:
Updated, Thanks

HaraldPanten commented:
@JasminSForgeFlow This PR is ready to rebased. On the other hand, you're modifying two different modules.

1- The migration of account_banking_sepa_direct_debit.
2- FIX for account_banking mandate.

Could you separate that in two different PR? One for the migration and the other one for the FIX?

THX

carolinafernandez-tecnativa committed:
[16.0][IMP] account_banking_sepa_direct_debit: Button send mandate per email

HaraldPanten commented:
Hi @MiquelRForgeFlow @JasminSForgeFlow Did you have availability to check my comments?

THX!

pedrobaeza commented:
Please rebase and fix CI, and include #1244

Commit:
[UPD] Update account_banking_sepa_direct_debit.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_sepa_direct_debit
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_sepa_direct_debit/

HaraldPanten commented:
/ocabot miggration account_banking_sepa_direct_debit

HaraldPanten commented:
/ocabot migration account_banking_sepa_direct_debit

JasminSForgeFlow committed:
[IMP] account_banking_sepa_direct_debit: black, isort, prettier

JasminSForgeFlow committed:
[MIG] account_banking_sepa_direct_debit: Migration to 17.0

JasminSForgeFlow commented:
> @JasminSForgeFlow This PR is ready to rebased. On the other hand, you're modifying two different modules.
> 
> 1- The migration of account_banking_sepa_direct_debit. 2- FIX for account_banking mandate.
> 
> Could you separate that in two different PR? One for the migration and the other one for the FIX?
> 
> THX

Done, Thanks

JasminSForgeFlow commented:
> Please rebase and fix CI, and include #1244

Done, Thanks

HaraldPanten commented:
LGTM @pedrobaeza 

Last tests should be directly in a production environment.

WDYT?

pedrobaeza commented on account_banking_sepa_direct_debit/models/mail_compose_message.py:
```diff
@@ -10,7 +11,7 @@ class MailComposeMessage(models.TransientModel):
-            auto_commit=auto_commit
-        )
+                self.env[wizard.model].sudo().browse(
+                    json.loads(wizard.res_ids)
```
Why do you need to convert it to JSON?

JasminSForgeFlow replied:
It is returning list in string format that's why

pedrobaeza reviewed (approved):
/ocabot merge nobump
